### PR TITLE
Update descriptor fields to resolve fft precision issue

### LINKED ISSUE #124096: MPS:  IRFFT returns incorrect results
### 🐛 Describe the bug

The FFT + IFFT on MPS is not as precise as CUDA or CPU.
With CUDA on Google Colab Pro:
```python
import torch

# Set the device to CUDA
device = torch.device("cuda")

# Create a real-valued signal
N = 64  # Define the size of the signal
signal = torch.rand(N, device=device)  # Random signal of size N on MPS device

# Perform FFT from real to complex
fft_result = torch.fft.rfft(signal)

# Perform inverse FFT to transform back to real space
ifft_result = torch.fft.irfft(fft_result, n=signal.shape[0])

# Print the output
print("Input Real Signal:", signal)
print("FFT Result:", fft_result)
print("Inverse FFT Result:", ifft_result)

# Calculate and display the error
error = torch.norm(signal - ifft_result)
print("Error between original and inverse FFT result:", error)

# Ensure correct alignment by comparing elements directly
print("Original and Inverse FFT are exactly the same:", torch.allclose(signal, ifft_result, atol=1e-6))
Input Real Signal: tensor([0.5556, 0.2825, 0.9749, 0.4927, 0.8606, 0.5090, 0.6194, 0.6863, 0.4115,
        0.7651, 0.5924, 0.3741, 0.8376, 0.7322, 0.1159, 0.7833, 0.6111, 0.4325,
        0.4651, 0.8310, 0.4474, 0.2033, 0.5109, 0.1103, 0.0299, 0.8218, 0.4031,
        0.2451, 0.8844, 0.8892, 0.8677, 0.3740, 0.5285, 0.7082, 0.9013, 0.9456,
        0.9304, 0.3395, 0.0222, 0.8953, 0.8396, 0.6433, 0.1134, 0.9663, 0.8216,
        0.8352, 0.5605, 0.9158, 0.7798, 0.9221, 0.0709, 0.6744, 0.2140, 0.1136,
        0.2181, 0.9840, 0.6949, 0.3716, 0.4990, 0.6069, 0.2475, 0.2762, 0.1909,
        0.1810], device='cuda:0')
FFT Result: tensor([ 3.5731e+01+0.0000j, -2.0241e+00+0.9830j, -3.9954e-02-3.6777j,
        -1.5710e+00-1.2591j,  1.5863e+00+0.7185j, -2.9887e+00-1.3917j,
        -1.5300e+00-2.0160j,  1.6517e+00-2.0552j, -1.9660e-01-1.8023j,
         9.0178e-01+1.6195j, -1.4341e+00-0.4520j,  6.9979e-01+0.0539j,
        -2.7289e+00-1.1149j,  2.8067e+00-0.9879j, -1.7785e+00+0.6767j,
         1.2185e+00-0.6068j,  2.5688e+00+1.2210j, -2.1503e+00+2.4158j,
        -1.2492e+00-2.5099j, -2.5199e+00+1.5649j,  2.0747e+00+0.1849j,
         2.5104e+00-2.1380j, -1.1488e+00+0.2445j,  9.8438e-03+0.3571j,
        -1.3884e+00+0.0272j,  1.1593e+00+2.2383j,  1.1816e+00+1.2621j,
        -5.0528e-01+0.5473j,  1.0649e+00+0.6632j,  4.4449e-01+0.5913j,
         3.5441e+00-0.0726j,  7.9013e-01+2.0291j, -2.0913e+00+0.0000j],
       device='cuda:0')
Inverse FFT Result: tensor([0.5556, 0.2825, 0.9749, 0.4927, 0.8606, 0.5090, 0.6194, 0.6863, 0.4115,
        0.7651, 0.5924, 0.3741, 0.8376, 0.7322, 0.1159, 0.7833, 0.6111, 0.4325,
        0.4651, 0.8310, 0.4474, 0.2033, 0.5109, 0.1103, 0.0299, 0.8218, 0.4031,
        0.2451, 0.8844, 0.8892, 0.8677, 0.3740, 0.5285, 0.7082, 0.9013, 0.9456,
        0.9304, 0.3395, 0.0222, 0.8953, 0.8396, 0.6433, 0.1134, 0.9663, 0.8216,
        0.8352, 0.5605, 0.9158, 0.7798, 0.9221, 0.0709, 0.6744, 0.2140, 0.1136,
        0.2181, 0.9840, 0.6949, 0.3716, 0.4990, 0.6069, 0.2475, 0.2762, 0.1909,
        0.1810], device='cuda:0')
Error between original and inverse FFT result: tensor(5.3909e-07, device='cuda:0')
Original and Inverse FFT are exactly the same: True
```
With CPU on INTEL iMac 27"
```
In [2]: import torch
   ...: 
   ...: # Check if MPS is available
   ...: if not torch.backends.mps.is_available():
   ...:     raise RuntimeError("MPS not available on this device")
   ...: 
   ...: # Set the device to MPS
   ...: device = torch.device("cpu")
   ...: 
   ...: # Create a real-valued signal
   ...: N = 64  # Define the size of the signal
   ...: signal = torch.rand(N, device=device)  # Random signal of size N on MPS device
   ...: 
   ...: # Perform FFT from real to complex
   ...: fft_result = torch.fft.rfft(signal)
   ...: 
   ...: # Perform inverse FFT to transform back to real space
   ...: ifft_result = torch.fft.irfft(fft_result, n=signal.shape[0])
   ...: 
   ...: # Print the output
   ...: print("Input Real Signal:", signal)
   ...: print("FFT Result:", fft_result)
   ...: print("Inverse FFT Result:", ifft_result)
   ...: 
   ...: # Calculate and display the error
   ...: error = torch.norm(signal - ifft_result)
   ...: print("Error between original and inverse FFT result:", error)
   ...: 
   ...: # Ensure correct alignment by comparing elements directly
   ...: print("Original and Inverse FFT are exactly the same:", torch.allclose(signal, ifft_result, atol=1e-6))
Input Real Signal: tensor([0.0499, 0.4680, 0.0302, 0.3498, 0.1693, 0.2607, 0.7666, 0.5321, 0.7172,
        0.0215, 0.3520, 0.3333, 0.7380, 0.7263, 0.3649, 0.4877, 0.7979, 0.3327,
        0.4125, 0.0322, 0.3502, 0.2421, 0.1081, 0.2191, 0.5429, 0.7382, 0.6048,
        0.6393, 0.2853, 0.4590, 0.9767, 0.0938, 0.9179, 0.9880, 0.1972, 0.5215,
        0.3056, 0.2720, 0.9675, 0.4114, 0.4247, 0.8357, 0.3912, 0.2396, 0.7974,
        0.2685, 0.2246, 0.4528, 0.9612, 0.9614, 0.8999, 0.7771, 0.6175, 0.9787,
        0.3101, 0.8608, 0.2468, 0.2307, 0.1346, 0.4056, 0.6062, 0.7231, 0.9320,
        0.1634])
FFT Result: tensor([31.2273+0.0000j, -1.3520+2.8785j, -1.3054+0.3543j, -3.5510-1.1144j,
         1.3571+0.8519j,  2.6081+0.2970j, -1.3665+2.6448j, -2.8771+2.7369j,
         1.1908+1.1168j, -1.3510+2.0015j,  0.5382+0.8023j, -2.0707-1.6480j,
         0.0988-2.5348j, -0.6662-0.8796j, -2.7091-0.1771j, -0.3838+1.2964j,
         0.8551-1.9871j,  0.2484+0.9134j, -1.9181-0.4970j, -0.2007-0.5681j,
         0.9216-0.9743j, -1.6056+0.0948j,  1.7320-4.1592j, -0.5929+2.6410j,
         0.3869-2.1391j,  0.3379+0.4569j, -0.8534-3.2982j, -1.9074-1.4680j,
         0.8035-1.5249j,  0.4705+0.3252j, -0.4477+1.8710j, -0.9944+1.5998j,
         1.1749+0.0000j])
Inverse FFT Result: tensor([0.0499, 0.4680, 0.0302, 0.3498, 0.1693, 0.2607, 0.7666, 0.5321, 0.7172,
        0.0215, 0.3520, 0.3333, 0.7380, 0.7263, 0.3649, 0.4877, 0.7979, 0.3327,
        0.4125, 0.0322, 0.3502, 0.2421, 0.1081, 0.2191, 0.5429, 0.7382, 0.6048,
        0.6393, 0.2853, 0.4590, 0.9767, 0.0938, 0.9179, 0.9880, 0.1972, 0.5215,
        0.3056, 0.2720, 0.9675, 0.4114, 0.4247, 0.8357, 0.3912, 0.2396, 0.7974,
        0.2685, 0.2246, 0.4528, 0.9612, 0.9614, 0.8999, 0.7771, 0.6175, 0.9787,
        0.3101, 0.8608, 0.2468, 0.2307, 0.1346, 0.4056, 0.6062, 0.7231, 0.9320,
        0.1634])
Error between original and inverse FFT result: tensor(3.6546e-07)
Original and Inverse FFT are exactly the same: True

```
With MPS on iMac 27" with AMD Radeon Pro 5700 XT:
```
% ipython
Python 3.10.13 (main, Sep 11 2023, 08:21:04) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.20.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import torch
   ...: 
   ...: # Check if MPS is available
   ...: if not torch.backends.mps.is_available():
   ...:     raise RuntimeError("MPS not available on this device")
   ...: 
   ...: # Set the device to MPS
   ...: device = torch.device("mps")
   ...: 
   ...: # Create a real-valued signal
   ...: N = 64  # Define the size of the signal
   ...: signal = torch.rand(N, device=device)  # Random signal of size N on MPS device
   ...: 
   ...: # Perform FFT from real to complex
   ...: fft_result = torch.fft.rfft(signal)
   ...: 
   ...: # Perform inverse FFT to transform back to real space
   ...: ifft_result = torch.fft.irfft(fft_result, n=signal.shape[0])
   ...: 
   ...: # Print the output
   ...: print("Input Real Signal:", signal)
   ...: print("FFT Result:", fft_result)
   ...: print("Inverse FFT Result:", ifft_result)
   ...: 
   ...: # Calculate and display the error
   ...: error = torch.norm(signal - ifft_result)
   ...: print("Error between original and inverse FFT result:", error)
   ...: 
   ...: # Ensure correct alignment by comparing elements directly
   ...: print("Original and Inverse FFT are exactly the same:", torch.allclose(signal, ifft_result, atol=1e-6))
/Users/davidlaxer/anaconda3/envs/AI-Feynman/lib/python3.10/site-packages/torch/library.py:168: UserWarning: Warning only once for all operators,  other operators may also be overrided.
  Overriding a previously registered kernel for the same operator and the same dispatch key
  operator: aten::var_mean.correction(Tensor self, int[1]? dim=None, *, Scalar? correction=None, bool keepdim=False) -> (Tensor, Tensor)
    registered at /Users/davidlaxer/pytorch/build/aten/src/ATen/RegisterSchema.cpp:6
  dispatch key: MPS
  previous kernel: registered at /Users/davidlaxer/pytorch/build/aten/src/ATen/RegisterCPU.cpp:31470
       new kernel: registered at /dev/null:1881 (Triggered internally at /Users/davidlaxer/pytorch/aten/src/ATen/core/dispatch/OperatorEntry.cpp:160.)
  self.m.impl(name, dispatch_key if dispatch_key != "" else "CompositeImplicitAutograd", fn)
Input Real Signal: tensor([0.8608, 0.5184, 0.1361, 0.4295, 0.3576, 0.1762, 0.6428, 0.4296, 0.1539,
        0.2445, 0.5251, 0.1419, 0.8278, 0.0490, 0.6466, 0.1222, 0.0596, 0.5996,
        0.6959, 0.9928, 0.9990, 0.9639, 0.7309, 0.7536, 0.6063, 0.3071, 0.5318,
        0.9431, 0.4022, 0.8120, 0.6358, 0.9048, 0.3648, 0.4482, 0.3339, 0.8373,
        0.6172, 0.3633, 0.4422, 0.2461, 0.1146, 0.1112, 0.3130, 0.0353, 0.0706,
        0.2076, 0.2046, 0.8289, 0.1582, 0.3371, 0.7776, 0.5222, 0.9413, 0.3857,
        0.4923, 0.3346, 0.0860, 0.5251, 0.3223, 0.4325, 0.0917, 0.6092, 0.8542,
        0.7759], device='mps:0')
FFT Result: tensor([30.3852+0.0000e+00j, -1.6317-2.9225e+00j,  1.3902+5.0550e+00j,
         0.5133-8.0040e-01j,  2.5358-2.9495e+00j,  0.8953+2.5815e+00j,
         1.9860+2.8584e+00j,  1.5863-8.8388e-01j, -2.1964+1.3933e+00j,
         1.4785+1.4994e+00j,  1.6552+8.6681e-01j,  2.2061-8.8684e-02j,
        -0.0527+2.0605e+00j, -1.0848+1.0977e-01j,  0.3218+4.0444e-01j,
        -0.0580-6.7318e-01j, -1.5737+2.0721e+00j, -0.3485-1.0628e+00j,
         0.5235+5.0354e-01j,  1.5206-4.3049e-01j,  0.6161-6.7625e-01j,
         0.2942+4.1318e-01j,  2.3510-2.1041e+00j,  0.4048-5.8936e-01j,
        -1.6097-6.3421e-01j, -1.8098-1.0714e+00j,  1.9137-4.2552e-03j,
         3.4070+4.7094e-02j, -1.1684+4.0498e-01j, -1.0623-4.2267e-01j,
        -2.0797+1.8722e+00j,  1.6259+9.6630e-01j, -0.3919+0.0000e+00j],
       device='mps:0')
Inverse FFT Result: tensor([0.8608, 0.7759, 0.8542, 0.6092, 0.0917, 0.4325, 0.3223, 0.5251, 0.0860,
        0.3346, 0.4923, 0.3857, 0.9413, 0.5222, 0.7776, 0.3371, 0.1582, 0.8289,
        0.2046, 0.2076, 0.0706, 0.0353, 0.3130, 0.1112, 0.1146, 0.2461, 0.4422,
        0.3633, 0.6172, 0.8373, 0.3339, 0.4482, 0.3648, 0.9048, 0.6358, 0.8120,
        0.4022, 0.9431, 0.5318, 0.3071, 0.6063, 0.7536, 0.7309, 0.9639, 0.9990,
        0.9928, 0.6959, 0.5996, 0.0596, 0.1222, 0.6466, 0.0490, 0.8278, 0.1419,
        0.5251, 0.2445, 0.1539, 0.4296, 0.6428, 0.1762, 0.3576, 0.4295, 0.1361,
        0.5184], device='mps:0')
Error between original and inverse FFT result: tensor(3.2877, device='mps:0')
Original and Inverse FFT are exactly the same: False

```
Could someone investigate?

### Versions

```
 % python collect_env.py
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: macOS 14.4.1 (x86_64)
GCC version: Could not collect
Clang version: 14.0.6
CMake version: version 3.22.1
Libc version: N/A

Python version: 3.10.13 (main, Sep 11 2023, 08:21:04) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Intel(R) Core(TM) i7-10700K CPU @ 3.80GHz

Versions of relevant libraries:
[pip3] audiolm-pytorch==0.0.1
[pip3] configmypy==0.1.0
[pip3] mypy==1.4.1
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.4
[pip3] onnxruntime==1.17.1
[pip3] optree==0.11.0
[pip3] pytorch-transformers==1.1.0
[pip3] tensorly-torch==0.4.0
[pip3] torch==2.2.0a0+gitca03f36
[pip3] torch-cluster==1.6.1
[pip3] torch-harmonics==0.6.5
[pip3] torch-scatter==2.1.1
[pip3] torch-sparse==0.6.17
[pip3] torch-spline-conv==1.2.2
[pip3] torch-struct==0.5
[pip3] torch-summary==1.4.5
[pip3] torch-utils==0.1.2
[pip3] torchaudio==2.2.0.dev20240227
[pip3] torchdata==0.7.1
[pip3] torchtext==0.17.2
[pip3] torchtraining-nightly==1604016577
[pip3] torchvision==0.18.0.dev20240227
[pip3] triton==2.1.0
[pip3] vector-quantize-pytorch==0.9.2
[conda] mkl                       2023.2.1                 pypi_0    pypi
[conda] nomkl                     3.0                           0  
[conda] numpy                     1.22.4                   pypi_0    pypi
[conda] numpy-base                1.26.4          py310hd8f4981_0  
[conda] optree                    0.11.0                   pypi_0    pypi
[conda] pytorch-transformers      1.1.0                    pypi_0    pypi
[conda] tensorly-torch            0.4.0                    pypi_0    pypi
[conda] torch                     2.3.0a0+gitc7e9c15          pypi_0    pypi
[conda] torch-cluster             1.6.1                    pypi_0    pypi
[conda] torch-harmonics           0.6.5                    pypi_0    pypi
[conda] torch-scatter             2.1.1                    pypi_0    pypi
[conda] torch-sparse              0.6.17                   pypi_0    pypi
[conda] torch-spline-conv         1.2.2                    pypi_0    pypi
[conda] torch-struct              0.5                      pypi_0    pypi
[conda] torch-summary             1.4.5                    pypi_0    pypi
[conda] torch-utils               0.1.2                    pypi_0    pypi
[conda] torchaudio                2.2.0.dev20240227          pypi_0    pypi
[conda] torchdata                 0.7.1                    pypi_0    pypi
[conda] torchtext                 0.17.2                   pypi_0    pypi
[conda] torchtraining-nightly     1604016577               pypi_0    pypi
[conda] torchvision               0.18.0.dev20240227          pypi_0    pypi
[conda] triton                    2.1.0                    pypi_0    pypi
[conda] vector-quantize-pytorch   0.9.2                    pypi_0    pypi
```

cc @ezyang @gchanan @zou3519 @kadeng @mruberry @peterbell10 @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

### PR DESCRIPTION
**Pull request by jhavukainen**
Fixes #124096

### CODE REVIEW

**Review by kulinseth — APPROVED** [review]

**Comment by kulinseth** [comment]
@jhavukainen can you add a test case as well ?

**Comment by jhavukainen** [comment]
> @jhavukainen can you add a test case as well ?

Done

**Review by malfet — APPROVED** [review]

File: test/test_mps.py [hunk 1]
```diff
@@ -2626,6 +2626,18 @@ def helper(input_shape, normalized_shape, eps=1e-05, elementwise_affine=True, dt
         # Regression test for https://github.com/pytorch/pytorch/issues/96113
         torch.nn.LayerNorm((16,), elementwise_affine=True).to("mps")(torch.randn(1, 2, 16).to("mps", dtype=torch.float16))
 
+    def test_ifft(self):
```
Comment (malfet) [inline test/test_mps.py:2629]:
```suggestion
    def test_ifft(self):
        if product_version < 14.0:
            raise unittest.skipTest("FFT is only supported on MacOS 14.0+")
```

**Comment by jhavukainen** [comment]
@malfet It looks like the cuda check is failing due to some server side issue that doesn't seem to be related to the PR. Is it OK to merge this?

**Pull request closed without merging**